app/src/main/java/io/github/pietrocaselani/moviestraker/tmdb/TMDB.kt

```diff
@@ -0,0 +1,63 @@
+package io.github.pietrocaselani.moviestraker.tmdb
+
+import com.google.gson.GsonBuilder
+import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import io.github.pietrocaselani.moviestraker.tmdb.services.Configuration
+import io.github.pietrocaselani.moviestraker.tmdb.services.Genres
+import io.github.pietrocaselani.moviestraker.tmdb.services.Movies
+import io.github.pietrocaselani.moviestraker.tmdb.services.Search
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+/**
+ * Created by pc on 24/06/17.
+ */
+open class TMDB(private val apiKey: String) {
+
+	companion object {
+		val API_HOST = "api.themoviedb.org"
+		val API_VERSION = "3"
+		val API_URL = "https://$API_HOST/$API_VERSION/"
+	}
+
+	private val retrofit: Retrofit by lazy {
+		retrofitBuilder().build()
+	}
+
+	protected open fun retrofitBuilder(): Retrofit.Builder {
```

Comment:
As I said before, the application can still extend the `TMDB` class and provide custom retrofit, okhttp or gson. Only the `api key` it's necessary to create an instance of `TMDB` and I'm providing the default implementation (witch manages the api key query parameter) for those components, but allowing to be extended, so it still a way to do dependecy injection.